maven/src/test/java/org/commonjava/maven/galley/maven/model/view/MavenPomViewTest.java

```diff
@@ -395,20 +395,28 @@ public void repositoriesWithUrlPropertyInProfile()
         MavenPomView pomView = loadPoms( "pom-with-repo-property-in-profile.xml" );
         List<RepositoryView> rvs = pomView.getAllRepositories();
 
-        for ( RepositoryView rv : rvs )
-        {
-            if ( rv.getName().equals( "repo.one" ) )
-            {
-                assertThat( rv.getUrl(), equalTo( "http://www.bar.com/repo" ) );
-            }
-            if ( rv.getName().equals( "test.oracle" ) )
-            {
-                assertThat( rv.getUrl(), equalTo( "http://test.oracle.repository" ) );
-            }
-            if ( rv.getName().equals( "test.two.oracle" ) )
-            {
-                assertThat( rv.getUrl(), equalTo( "http://test.two.oracle.repository" ) );
-            }
-        }
+        assertThat( rvs.get( 0 ).getName(), equalTo( "repo.one" ) );
+        assertThat( rvs.get( 0 ).getUrl(), equalTo( "http://repo.one.repository" ) );
+        assertThat( rvs.get( 1 ).getName(), equalTo( "test.oracle.repo" ) );
+        assertThat( rvs.get( 1 ).getUrl(), equalTo( "http://test.oracle.repository" ) );
+        assertThat( rvs.get( 2 ).getName(), equalTo( "test.second.oracle.repo" ) );
+        assertThat( rvs.get( 2 ).getUrl(), equalTo( "http://another.test.two.oracle.repository" ) );
+    }
+
+    @Test
+    public void resolveExpressionsBothInDepAndProfile()
```

Comment:
Hi, @jdcasey

Sorry I'm just back from the long holiday, and since limited in vpn, so didn't check your email in time.

I was pushing another commit for this, pls check.

And about your concern on test method: resolveExpressionsBothInDepAndProfile, before using RecursionInterceptor, some part of the tests shoud be passed, and for some test cases like redeclare the properties both in main and profile, it seems not passed IIRC, it will be failed into a wrong sistuation when using ssi to do some iteration, not all the correct active-profiles' profile-properties will be retrieved for current repository expression.

Anyway, after using the RecursionInterceptor, I try to cover all the test cases in the pom file including your comments' cases, and they all pass and run in the correct way. So I update this test and add all the test cases that I could cover into the test methods as you see.